template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java

```diff
@@ -0,0 +1,106 @@
+
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmHostGroupRoleConfigProvider;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.views.HostgroupView;
+
+@Component
+public class CruiseControlRoleConfigProvider implements CmHostGroupRoleConfigProvider {
+
+    private static final String DEFAULT_GOALS = "default.goals";
+
+    private static final String HARD_GOALS = "hard.goals";
+
+    private static final String ANOMALY_DETECTION_GOALS = "anomaly.detection.goals";
+
+    private static final String SELF_HEALING_ENABLED = "self.healing.enabled";
+
+    private static final String ANOMALY_NOTIFIER_CLASS = "anomaly.notifier.class";
+
+    private static final String METRIC_ANOMALY_FINDER_CLASS = "metric.anomaly.finder.class";
+
+    private static final String CPU_BALANCE_THRESHOLD = "cpu.balance.threshold";
+
+    private static final String DISK_BALANCE_THRESHOLD = "disk.balance.threshold";
+
+    private static final String NETWORK_INBOUND_BALANCE_THRESHOLD = "network.inbound.balance.threshold";
+
+    private static final String NETWORK_OUTBOUND_BALANCE_THRESHOLD = "network.outbound.balance.threshold";
+
+    private static final String REPLICA_COUNT_BALANCE_THRESHOLD = "replica.count.balance.threshold";
+
+    private static final String CPU_CAPACITY_THRESHOLD = "cpu.capacity.threshold";
+
+    private static final String DISK_CAPACITY_THRESHOLD = "disk.capacity.threshold";
+
+    private static final String NETWORK_INBOUND_CAPACITY_THRESHOLD = "network.inbound.capacity.threshold";
+
+    private static final String NETWORK_OUTBOUND_CAPACITY_THRESHOLD = "network.outbound.capacity.threshold";
+
+    private static final String LEADER_REPLICA_COUNT_BALANCE_THRESHOLD = "leader.replica.count.balance.threshold";
+
+    @Override
+    public String getServiceType() {
+        return CruiseControlRoles.CRUISECONTROL;
+    }
+
+    @Override
+    public Set<String> getRoleTypes() {
+        return Set.of(CruiseControlRoles.CRUISE_CONTROL_SERVER);
+    }
+
+    @Override
+    public boolean sharedRoleType(String roleType) {
+        return false;
+    }
+
+    @Override
+    public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, HostgroupView hostGroupView, TemplatePreparationObject source) {
+        String cdpVersion = source.getBlueprintView().getProcessor().getStackVersion() == null ?
+                "" : source.getBlueprintView().getProcessor().getStackVersion();
+
+        if (isVersionNewerOrEqualThanLimited(cdpVersion, CLOUDERA_STACK_VERSION_7_2_11)) {
+            return List.of(config(SELF_HEALING_ENABLED, "true"),
+                    config(ANOMALY_NOTIFIER_CLASS, "com.linkedin.kafka.cruisecontrol.detector.notifier.SelfHealingNotifier"),
```

Comment:
I think all of this should be a default variable which is setted on CM side.

Reply:
We have different default configs for Private Cloud and DataHub. Private Cloud uses the upstream defaults which are the CM defaults as well and DataHub uses customized ones. This is why we need this config provider.